src/gig_argument.c

```diff
@@ -1182,9 +1187,24 @@ c_garray_to_scm(C2S_ARG_DECL)
     _arg.v_pointer = array->data;
     _meta.is_raw_array = TRUE;
     _meta.length = array->len;
+    // The GArray retains ownership of the conents.
+    _meta.transfer = GI_TRANSFER_NOTHING;
 
     size = array->len;
+
     c_native_array_to_scm(subr, argpos, &_meta, &_arg, object, size);
+
+    // Since the GArray retained ownership of the contents, we free as
+    // necessary here.
+    if (meta->transfer == GI_TRANSFER_EVERYTHING) {
+        if (meta->params[0].gtype == G_TYPE_STRING ||
+            meta->params[0].gtype == G_TYPE_LOCALE_STRING)
```

Comment:
G_TYPE_LOCALE_STRING no longer exists on master, this case is already caught by gtype == G_TYPE_STRING.